resolwe/process/parser.py

```diff
@@ -111,9 +113,66 @@ def __init__(self, source):
         self.source = source
         self.processes = []
         self.base_classes = set()
+        # Mapping betwwen process slug and AST node representing the class.
+        self.ast_nodes = dict()
 
         super().__init__()
 
+    def _get_called_slug(self, process_slug, called_node: ast.Call):
+        """Get the slug of the process being called.
+
+        :raises RuntimeError: when slug could not be determined (for instance when
+            the slug is being a variable).
+        """
+        # Both methods (run_process and get_latest) take argument slug as the first
+        # argument, so no separation is necessary.
+        argument_names = ("slug", "inputs")
+        arguments = [arg for arg in called_node.args]
+        all_arguments = {
+            name: value for name, value in zip_longest(argument_names, arguments)
```

Comment:
This will work in most use-cases, but note that `run_process(inputs={...}, slug='...')` is also a valid syntax.

Reply:
Will this not work as expected? The zip_longest will be empty (only keyword arguments are given) and all_arguments will be empty after the first call, but the next one will update it with the keyword arguments.